Sources/AsyncHTTPClient/Utils.swift

```diff
@@ -32,3 +33,36 @@ public final class HTTPClientCopyingDelegate: HTTPClientResponseDelegate {
         return ()
     }
 }
+
+extension ClientBootstrap {
+    static func makeHTTPClientBootstrapBase(group: EventLoopGroup, host: String, port: Int, configuration: HTTPClient.Configuration, channelInitializer: ((Channel) -> EventLoopFuture<Void>)? = nil) -> ClientBootstrap {
+        return ClientBootstrap(group: group)
+            .channelOption(ChannelOptions.socket(SocketOptionLevel(IPPROTO_TCP), TCP_NODELAY), value: 1)
+
+            .channelInitializer { channel in
+                let channelAddedFuture: EventLoopFuture<Void>
+                switch configuration.proxy {
+                case .none:
+                    channelAddedFuture = group.next().makeSucceededFuture(())
+                case .some:
+                    channelAddedFuture = channel.pipeline.addProxyHandler(host: host, port: port, authorization: configuration.proxy?.authorization)
+                }
+                return channelAddedFuture.flatMap { (_: Void) -> EventLoopFuture<Void> in
+                    channelInitializer?(channel) ?? group.next().makeSucceededFuture(())
+                }
+            }
+    }
+}
+
+func resolve(timeout: TimeAmount?, deadline: NIODeadline?) -> TimeAmount? {
```

Comment:
I think it might be better to put this function inside Timeout struct (namespace is basically), what do you think? Its difficult to know where a function comes from (without an IDE) otherwise

Reply:
@artemredkin It would be better indeed, thank you!

Reply:
@artemredkin I tried this and it turns out this requires calls with `HTTPClient.Configuration.Timeout.resolve(...)` which is quite big, what do you think? Maybe we should put this on `HTTPClient` instead?